### ISSUE TITLE
Unprintable Characters

### ISSUE THREAD

**Issue by lincomatic** [mapsforge/mapsforge#812]
I am building maps with selectable languages. I have found that some of the characters are unprintable. I don't know if this is a Java issue or a typeface issue. I have compared the string data in my maps to the original OSM XML source, and the mapwriter process is definitely not the culprit.. the strings in the Mapsforge map are identical to the OSM XML input data. Below are some examples:

![image](https://cloud.githubusercontent.com/assets/1422186/14063732/00f1d45c-f393-11e5-847d-19e2a6a82028.png)

![image](https://cloud.githubusercontent.com/assets/1422186/14063733/288c9c5e-f393-11e5-908d-574752f341de.png)

For reference, one of the unprintable strings is the name of Cambodia in its native script. Below is the OSM data from which it was generated. The name that I am trying to print on the screen is the default "name" tag:

<?xml version='1.0' encoding='UTF-8'?>
<osm version="0.6" generator="Osmosis 0.43.1">
  <bounds minlon="104.80000" minlat="13.50000" maxlon="104.90000" maxlat="13.70000" origin="http://www.openstreetmap.org/api/0.6"/>
  <node id="432424952" version="-1" timestamp="1969-12-31T23:59:59Z" changeset="-1" lat="13.5066394" lon="104.869423">
    <tag k="name" v="កម្ពុជា"/>
    <tag k="place" v="country"/>
    <tag k="name:af" v="Kambodja"/>
    <tag k="name:ak" v="Kambodia"/>
    <tag k="name:am" v="ካምቦዲያ"/>
    <tag k="name:an" v="Cambocha"/>
    <tag k="name:ar" v="كمبوديا"/>

I suspect that the problem is the typeface that Mapsforge is using? Mapnik is able to render these strings correctly, as can be seen on the openstreetmaps.org website.

**Comment by lincomatic** [comment]
Hmm.. some of the UTF-8 characters in my previous message are unprintable on the github website. Here is a link to the current OSM version of the data for Cambodia.

http://www.openstreetmap.org/node/432424952

Note that it's slightly different from the version that I used to generate my map. "(Cambodia)" has been appended to the unprintable characters.

**Comment by devemux86** [comment]
I made a quick test with Bangladesh from Geofabrik, showing on Android and Java (Linux).

![android](https://cloud.githubusercontent.com/assets/3484020/14064305/2ba3e8c4-f407-11e5-8908-cd07470d8e60.png)
![java](https://cloud.githubusercontent.com/assets/3484020/14064307/301324ba-f407-11e5-86de-3809f5234f66.png)

**Comment by ludwigb** [comment]
It is also that on certain devices certain character sets are not installed
to save space (a vendor decision). Rendering on device obviously requires
the fonts (or fallback fonts) to be installed.

**Comment by lincomatic** [comment]
OK, it seems that maybe the issue is indeed system dependent. I also created a Bangladesh map from Geofabrik data, and I am still getting the empty boxes. I am running Oracle JDK 1.7 on Windows 8.1. I did some reading up on Java, and it appears that it uses the OS's fonts. I have Bangla fonts installed, and the Bengali characters display correctly in my native text editor, but they don't display under Mapsforge Java apps.

**Comment by lincomatic** [comment]
I am closing this... it is clearly just a Java configuration issue on my part.

**Comment by applantation** [comment]
Please still post your solution to this. While this is not a mapsforge
issue, your work will still help others running into the same situation.

Thanks

**Comment by lincomatic** [comment]
I tried to figure out how to modify the JRE’s fontconfig.properties file, but the documentation is confusing, and I was not having much luck, so I decided to take an easy way out. I copied the /system/fonts directory from my Android machine, which was displaying most of the characters properly, to <JRE>/lib/fonts/fallback. There was only one font missing, Dzongkha, which is used for Bhutan. I downloaded it from http://www.dzongkha.gov.bt/IT/Font_Installer_Instructions.en.html

If anyone is interested in which fonts I’m using I can post a list. Some of them are obviously needed, but I am sure that I also copied some useless fonts.
